File: backends/spring-kotlin/src/main/kotlin/com/example/todo/controller/CategoryController.kt
package com.example.todo.controller

import com.example.todo.dto.CategoryRequest
import com.example.todo.dto.CategoryResponse
import com.example.todo.service.CategoryService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/categories")
class CategoryController(
    private val categoryService: CategoryService
) {
    @GetMapping
    fun getCategories(): ResponseEntity<List<CategoryResponse>> {
        return ResponseEntity.ok(categoryService.getAllCategories())
    }

    @PostMapping
    fun createCategory(@RequestBody request: CategoryRequest): ResponseEntity<CategoryResponse> {
        val created = categoryService.createCategory(request)
        return ResponseEntity.status(HttpStatus.CREATED).body(created)
    }

    @DeleteMapping("/{id}")
    fun deleteCategory(@PathVariable id: Long): ResponseEntity<Void> {
        categoryService.deleteCategory(id)
        return ResponseEntity.noContent().build()
    }
}

File: backends/nestjs/src/category/category.controller.ts
import { Controller, Get, Post, Delete, Body, Param, HttpCode } from '@nestjs/common';
import { CategoryService } from './category.service';
import { CategoryRequest } from './dto/category-request.dto';

@Controller('api/categories')
export class CategoryController {
  constructor(private readonly categoryService: CategoryService) {}

  @Get()
  async getCategories() {
    return this.categoryService.findAll();
  }

  @Post()
  @HttpCode(201)
  async createCategory(@Body() request: CategoryRequest) {
    return this.categoryService.create(request);
  }

  @Delete(':id')
  @HttpCode(204)
  async deleteCategory(@Param('id') id: string) {
    await this.categoryService.remove(Number(id));
    return;
  }
}
